pilot/cmd/pilot-agent/status/server/server.go

```diff
@@ -355,6 +329,24 @@ func (s *Server) Run(ctx context.Context) {
 	mux.HandleFunc("/debug/pprof/trace", s.handlePprofTrace)
 	mux.HandleFunc("/debug/ndsz", s.handleNdsz)
 
+	var handler http.Handler = mux
+	if s.config.DebugTapClient != nil {
+		debugTapProxy := debugtap.NewProxy(s.config.DebugTapClient)
+
+		// Registers HTTP Handlers for Debug Tap
+		debugTapProxy.RegisterHTTPHandler(mux)
+		grpcs := grpc.NewServer(istiogrpc.ServerOptions(istiokeepalive.DefaultOption())...)
+		debugTapProxy.RegisterGRPCHandler(grpcs)
+		mixedHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
do we need the `isRequestFromLocalhost` stuff?

Reply:
Good point. Done.